File: frontend/src/data/personas.ts
export type LearningPreference = 'videos' | 'reading' | 'flashcards' | 'hands-on' | 'ai-interaction' | 'quizzes'

export type Persona = {
  id: string
  name: string
  major: string
  courseTitle: string
  courseUrl: string
  hoursPerWeek: number
  learningPreferences: LearningPreference[]
  learningStyle: string
  experienceLevel: 'beginner' | 'intermediate' | 'advanced'
  primaryRoadmapPath: string
}

export const PERSONAS: Record<string, Persona> = {
  alice: {
    id: 'alice',
    name: 'Alice',
    major: 'Finance & Data Science',
    courseTitle: 'Introduction to Computer Science and Programming in Python',
    courseUrl: 'https://ocw.mit.edu/courses/6-0001-introduction-to-computer-science-and-programming-in-python-fall-2016/',
    hoursPerWeek: 5,
    learningPreferences: ['videos', 'hands-on'],
    learningStyle: 'Prefers video lectures and hands-on coding problems. Not very familiar with programming yet.',
    experienceLevel: 'beginner',
    primaryRoadmapPath: '/roadmap/python',
  },
  bob: {
    id: 'bob',
    name: 'Bob',
    major: 'Business',
    courseTitle: 'Financing Economic Development',
    courseUrl: 'https://ocw.mit.edu/courses/11-437-financing-economic-development-fall-2016/',
    hoursPerWeek: 2,
    learningPreferences: ['reading', 'ai-interaction'],
    learningStyle: 'Enjoys reading detailed materials and discussing topics with AI using lots of examples. Very familiar with the subject matter.',
    experienceLevel: 'advanced',
    primaryRoadmapPath: '/roadmap/financing',
  },
  charles: {
    id: 'charles',
    name: 'Charles',
    major: 'Accounting',
    courseTitle: 'Introduction to Financial and Managerial Accounting',
    courseUrl: 'https://ocw.mit.edu/courses/15-501-introduction-to-financial-and-managerial-accounting-spring-2004/',
    hoursPerWeek: 10,
    learningPreferences: ['flashcards', 'quizzes'],
    learningStyle: 'Loves flashcard reviews and knowledge-testing questions. Can dedicate significant time to studying.',
    experienceLevel: 'intermediate',
    primaryRoadmapPath: '/roadmap/accounting',
  },
}

export const DEFAULT_PERSONA = 'charles'

File: backend/personas.py
from __future__ import annotations

PERSONAS: dict[str, dict] = {
    "alice": {
        "student_id": "a0000001-0000-4000-8000-000000000001",
        "name": "Alice",
        "major": "Finance and Data Science",
        "course": "python",
        "familiarity": "beginner",
        "learning_style": "videos and hands-on problems",
        "hours_per_week": 5,
        "notes": (
            "Alice is not very familiar with coding. Prefer simple language, "
            "relatable analogies, and hands-on examples. Avoid heavy jargon. "
            "Keep explanations approachable and build intuition before diving into details."
        ),
    },
    "bob": {
        "student_id": "b0000002-0000-4000-8000-000000000002",
        "name": "Bob",
        "major": "Business",
        "course": "financing",
        "familiarity": "high",
        "learning_style": "reading and interacting with AI, lots of real-world examples",
        "hours_per_week": 2,
        "notes": (
            "Bob is already very familiar with the topic. Skip basic definitions and "
            "go straight to nuance, edge cases, and real-world business applications. "
            "Use dense, example-rich content. He has limited time so keep it efficient."
        ),
    },
    "charles": {
        "student_id": "c0000003-0000-4000-8000-000000000003",
        "name": "Charles",
        "major": "Accounting",
        "course": "accounting",
        "familiarity": "intermediate",
        "learning_style": "flashcard reviews and questions to test knowledge",
        "hours_per_week": 10,
        "notes": (
            "Charles wants to be tested frequently. Emphasize knowledge checks, "
            "key term definitions, and structured review. He can handle more depth "
            "and appreciates thorough explanations followed by questions."
        ),
    },
}


def get_persona(persona_id: str) -> dict:
    persona = PERSONAS.get(persona_id.lower())
    if not persona:
        raise ValueError(f"Unknown persona '{persona_id}'. Choose from: {list(PERSONAS.keys())}")
    return persona
